### PR TITLE
Image resizing

### PR DESCRIPTION
**Pull request by CGCooke**
Added a command line option to convert images to thumbnails on startup.

This helps to reduce the memory requirements where there is a large dataset with large images.

Example usage:
`python -m bulk image my_dataset.csv --thumbnail`

### CODE REVIEW

**Comment by koaning** [comment]
I'm not 100% sure if I'd place it there. 

Maybe instead have it live in `python -m bulk utils resize`? 

I also wonder if this is a batch job you'd want to run before running `python -m bulk image`? Doesn't it become super slow to do resizing before loading it into bulk?

**Comment by koaning** [comment]
Also, a unit test would be nice 🙂

**Comment by CGCooke** [comment]
- Maybe instead have it live in python -m bulk utils resize?
This sounds like a good idea,


- I also wonder if this is a batch job you'd want to run before running python -m bulk image? Doesn't it become super slow to do resizing before loading it into bulk?

It is slower, however, once you get up to a certain scale, (thousands of HD images), the dataset will not load at all, without being resized.

I think it should be a batch job, and then place the thumbnails into a thumbnail folder?

**Comment by koaning** [comment]
>  into a thumbnail folder?

I'd require the user to pass a folder explicitly. And if said folder already exists then the user should be prompted to confirm.

**Comment by CGCooke** [comment]
"And if said folder already exists then the user should be prompted to confirm.",
I'm not sure how to do this, but I've updated the code on the branch with the feedback.

File: README.md [hunk 1]
```diff
@@ -132,6 +132,21 @@ This generates a csv file that can be loaded in bulk via;
...
+You can also generate a set of thumbnails for your images. This can be useful if you're working with a large dataset. 
+
+```
+python -m bulk util resize ready.csv temp   
+```
+
+This will create a folder called `temp` with all the resized images. You can then use this folder as the `--thumbnail-path` argument.
+
+```
+python -m bulk image ready.csv --thumbnail-path temp
```
Comment (koaning) [inline README.md:145]: Can't the `ready.csv` file have path that point to the thumbnail path? It occurs to me that it is the users responsibility to point to the right image path and to provide the correct x,y coordinates for them.

File: bulk/cli/util.py [hunk 2]
```diff
@@ -25,3 +30,20 @@ def concat(
...
         df = df.sample(frac=1)
     df.drop_duplicates().to_csv(out, index=False)
+
+@app.command("resize")
+def resize(
+        path: pathlib.Path = typer.Argument(..., help="Path to .csv/.jsonl file", exists=True),
+        thumbnail_paths: pathlib.Path = typer.Argument(..., help="Path to folder to store thumbnails.", exists=False)):
+    """Resize the images into thumbnails."""
+
+    os.system(f"mkdir {thumbnail_paths}")
```
Comment (koaning) [inline bulk/cli/util.py:46]: I prefer to use `Pathlib` here, but that's also something I can do post-PR.

**Review by koaning — CHANGES_REQUESTED** [review]
Left one comment on the `--thumbnail-path` mechanic that should be discussed/adressed. 

Also, would it be possible to add a small unit test for this? Maybe just a smoke test?

File: bulk/cli/util.py (hunk 2 above)
Comment (koaning) [reply bulk/cli/util.py:46]: Same with the comment on the prompt, by the way. That's also something I can pickup.

**Comment by koaning** [comment]
If you prefer, I can wrap up this PR on my side. I'm planning on switching command line backends anyway and that way you can still get your feature in, it's just that I'll wrap it up sometime later this week.

**Comment by CGCooke** [comment]
> If you prefer, I can wrap up this PR on my side. I'm planning on switching command line backends anyway and that way you can still get your feature in, it's just that I'll wrap it up sometime later this week.

Thanks, that's really appreciated. I was having a little trouble getting the testing to work properly.

**Comment by koaning** [comment]
No worries. I'll pick this up sometime later, but I'll make sure it gets included somehow!

**Pull request merged**